src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ReferralService.kt

```diff
@@ -452,6 +466,39 @@ class ReferralService(
 
     return newDetails
   }
+  private fun proccessChangeLog(ammendTopic: AmendTopic, oldValue: ReferralAmendmentDetails, newValue: ReferralAmendmentDetails, referral: Referral, actor: AuthUser, update: UpdateReferralDetailsDTO) {
+    val changelog = Changelog(
+      referral.id,
+      UUID.randomUUID(),
+      ammendTopic,
+      oldValue,
+      newValue,
+      update.reasonForChange,
+      OffsetDateTime.now(),
+      actor
+    )
+
+    changelogRepository.save(changelog)
+  }
+
+  private fun handleChangLog(topic: AmendTopic, referral: Referral, update: UpdateReferralDetailsDTO, actor: AuthUser) {
+    when (topic) {
+      AmendTopic.COMPLETION_DATETIME -> {
+        val completionDeadline = referral?.completionDeadline
+        val oldValue = ReferralAmendmentDetails(listOf(completionDeadline.toString()))
+        val newValue = ReferralAmendmentDetails(listOf(update.completionDeadline.toString()))
+        proccessChangeLog(AmendTopic.COMPLETION_DATETIME, oldValue, newValue, referral, actor, update)
```

Comment:
Minor comment-could just pass the referral id instead of the whole referral object